sql/core/src/main/scala/org/apache/spark/sql/expressions/WindowSpec.scala

```diff
@@ -174,28 +191,22 @@ class WindowSpec private[sql](
    */
   // Note: when updating the doc for this method, also update Window.rangeBetween.
   def rangeBetween(start: Long, end: Long): WindowSpec = {
```

Comment:
It might be the easiest to make it take a `Column`, so `rangeBetween(begin: Column, end: Column)`, only downside to this is that we need some way to express the special boundaries (`current row`, `unbounded`). Also cc @rxin